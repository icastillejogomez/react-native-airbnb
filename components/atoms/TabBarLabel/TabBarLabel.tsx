import { AirbnbText } from '@/components/native'
import { FC } from 'react'
import { StyleSheet } from 'react-native'

export type TabBarLabelProps = {
  testID?: string
  isFocused: boolean
  label: string
}

const TabBarLabel: FC<TabBarLabelProps> = ({ isFocused, label, testID }) => {
  return (
    <AirbnbText
      testID={testID}
      variant="caption"
      weight={isFocused ? '600' : '400'}
      color={isFocused ? 'primary-core' : 'primary'}
      style={styles.label}>
      {label}
    </AirbnbText>
  )
}
TabBarLabel.displayName = 'TabBarLabel'
export { TabBarLabel }

const styles = StyleSheet.create({
  label: {
    fontSize: 10,
  },
})
