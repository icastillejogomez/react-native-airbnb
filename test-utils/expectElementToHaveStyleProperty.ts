import { ReactTestInstance } from 'react-test-renderer'

const reduceStyles =
  (styleProperty: string) =>
  (previousValue: string | undefined, style: Record<string, any> | null | false | undefined): string | number | undefined => {
    if (Array.isArray(style)) {
      return style.reduce(reduceStyles(styleProperty), undefined) as string | undefined
    }
    if (style && styleProperty in style) {
      previousValue = style[styleProperty] as string
    }
    return previousValue
  }

export function expectElementToHaveStyleProperty(element: ReactTestInstance, styleProperty: string, value: any) {
  const style = element.props.style

  expect(style).toBeDefined()
  if (!Array.isArray(style)) {
    return expect(style![styleProperty]).toEqual(value)
  }

  expect(Array.isArray(style)).toBeTruthy()
  const styleValue = style!.reduce(reduceStyles(styleProperty), undefined)

  expect(styleValue).toBe(value)
}
