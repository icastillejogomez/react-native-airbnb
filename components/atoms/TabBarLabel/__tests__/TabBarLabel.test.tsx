import { render } from '@testing-library/react-native'
import { TabBarLabel } from '../TabBarLabel'
import { expectElementToHaveStyleProperty } from '@/test-utils'

describe('TabBarLabel', () => {
  it('should render', () => {
    render(<TabBarLabel label={'Explore'} isFocused={true} />)
  })

  it('should match snapshot', () => {
    expect(render(<TabBarLabel label={'Explore'} isFocused={true} />)).toMatchSnapshot()
  })

  it('should render 10 pixels font size', () => {
    const tree = render(<TabBarLabel testID="label" label={'Explore'} isFocused={true} />)
    const $elements = tree.getAllByTestId('label')
    expect(Array.isArray($elements)).toBe(true)
    expect($elements).toHaveLength(2)
    const $element = $elements.at(-1)!
    expectElementToHaveStyleProperty($element, 'fontSize', 10)
  })
})
