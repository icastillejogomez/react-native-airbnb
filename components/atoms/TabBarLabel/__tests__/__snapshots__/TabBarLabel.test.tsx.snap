// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarLabel should match snapshot 1`] = `
<RNCMaskedView>
  <View
    pointerEvents="none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 16,
          },
          {},
          {
            "fontFamily": "Montserrat-Medium",
            "fontSize": 12,
            "lineHeight": 20,
          },
          {
            "fontFamily": "Montserrat-SemiBold",
          },
          {},
          {},
          {
            "textAlign": "auto",
          },
          {
            "fontSize": 10,
          },
          {},
        ]
      }
    >
      Explore
    </Text>
  </View>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        4293271117,
        4293074015,
        4292281446,
      ]
    }
    endPoint={
      [
        1,
        0.5,
      ]
    }
    locations={
      [
        0,
        0.5,
        1,
      ]
    }
    startPoint={
      [
        0,
        0.5,
      ]
    }
  >
    <Text
      style={
        [
          [
            {
              "fontSize": 16,
            },
            {},
            {
              "fontFamily": "Montserrat-Medium",
              "fontSize": 12,
              "lineHeight": 20,
            },
            {
              "fontFamily": "Montserrat-SemiBold",
            },
            {},
            {},
            {
              "textAlign": "auto",
            },
            {
              "fontSize": 10,
            },
            {},
          ],
          {
            "opacity": 0,
          },
        ]
      }
    >
      Explore
    </Text>
  </ViewManagerAdapter_ExpoLinearGradient>
</RNCMaskedView>
`;
